### PR TITLE
Fix: Deprecate or promote WARN level in logging package

### LINKED ISSUE #4373: Deprecate or promote WARN level in logging package
Although used sparingly, our logging implementation supports a WARN level. This is however currently not possible to set as the `--log-level` of the server, and the default implementation of `.Warn` currently just [defers](https://github.com/open-policy-agent/opa/blob/main/logging/logging.go#L145) to log at ERROR level, making it somewhat confusing/misleading.

I'd suggest either deprecate/remove the WARN level and rewrite the few places where it's used to log at INFO or ERROR, or that we promote WARN to a distinct logging level available as a `--log-level` option.

### PR DESCRIPTION
**Pull request by damienjburks**
Fixes #4373 

This is a new PR that was raised. The last PR was closed because of merging issues.
Old PR: #4507

### CODE REVIEW

File: logging/logging.go [hunk 1]
```diff
@@ -87,9 +87,7 @@ func (l *StandardLogger) getFields() map[string]interface{} {
 func (l *StandardLogger) SetLevel(level Level) {
 	var logrusLevel logrus.Level
 	switch level {
-	case Error:
-		logrusLevel = logrus.ErrorLevel
-	case Warn:
```
Comment (anderseknert) [inline logging/logging.go:92]: While it's not possible to set the log level to `Warn` via CLI options, I think one may still do this via the Go API.. so we might want to keep this case as well.

File: logging/logging_test.go [hunk 2]
```diff
@@ -19,6 +19,15 @@ func TestWithFields(t *testing.T) {
 	}
 }
 
+func TestWithWarnAsError(t *testing.T) {
```
Comment (anderseknert) [inline logging/logging_test.go:22]: This currently does not involve `Warn` :)
Can we add a test that asserts that when set to `Error`, we still see `Warn` logs in the output?

Comment (damienjburks) [reply logging/logging_test.go]: For sure. I just committed some of my latest changes to test for it. Check it out.

File: logging/logging_test.go [hunk 3]
```diff
@@ -19,6 +21,26 @@ func TestWithFields(t *testing.T) {
...
 }
 
+func TestCaptureWarningWithErrorSet(t *testing.T) {
+	buf := bytes.Buffer{}
+	logger := New()
+	logger.SetOutput(&buf)
+	logger.SetLevel(Error)
+
+	logger.Warn("This is a warning. Next time, I won't compile.")
+	logger.Error("Fix your issues. I'm not compiling.")
```
Comment (anderseknert) [inline logging/logging_test.go:31]: 😄

**Review by anderseknert — APPROVED** [review]
LGTM! Thanks a lot for bearing with me! 👍

**Pull request merged**